components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -778,11 +784,19 @@ private String getRefreshToken(OAuthTokenReqMessageContext tokReqMsgCtx, OauthTo
     }
 
     private void setRefreshTokenDetailsFromExistingToken(AccessTokenDO existingAccessTokenDO,
-                                                         AccessTokenDO newTokenBean) {
+                                                         AccessTokenDO newTokenBean,
+                                                         long refreshTokenExpiryInMillisFromActions) {
```

Comment:
Why not pass the message context and do and better not say actions here, rather it gives precendece for message context

Reply:
And we can add a comment that actions is a place where message context gets updated

Reply:
Fixed 49caebcb5cca897a855b8ca05cb5db3e23952e9d